docs/manual/called-methods-checker.tex

```diff
@@ -0,0 +1,187 @@
+construct objects, but it is error-prone.  For example, failing to provide
+a required argument causes a run-time error that manifests during testing
+or in the field, instead of at compile time as for regular Java
+constructors.  The Called Methods Checker verifies at compile time that
+your code correctly uses the builder pattern, never omitting a required
+argument. The checker has built-in support for
+\href{https://projectlombok.org/}{Lombok} (see the caveats about Lombok in
+Section~\ref{called-methods-lombok}) and
+\href{https://github.com/google/auto/blob/master/value/userguide/index.md}{AutoValue}.
+
+You can verify other builders, or verify other properties of the form
+``foo() must be called before bar()'', by writing method specifications.
+Section~\ref{called-methods-other} describes another example related to a
+security property.
+
+If the checker issues no warnings, then you have a guarantee that your code
+supplies all the required information to the builder.  The checker might
+yield a false positive warning when your code is too tricky for it to
+verify.  Please submit an
+\href{https://github.com/typetools/checker-framework/issues}{issue} if you
+discover this.
+
+
+\sectionAndLabel{How to run the Called Methods Checker}{called-methods-run-checker}
+
+\begin{Verbatim}
+javac -processor CalledMethods MyFile.java ...
+\end{Verbatim}
+
+
+\sectionAndLabel{For Lombok users}{called-methods-lombok}
+
+The Called Methods Checker supports projects that use Lombok via
+the \href{https://plugins.gradle.org/plugin/io.freefair.lombok}{io.freefair.lombok} Gradle plugin automatically.
+However, note that the checker's error messages refer to Lombok's output, which is a variant of your source code
+that appears in a \<delombok> directory.
+To fix issues, you should edit your original source code, \textbf{not} the files in the checker's error messages.
+
+If you use Lombok with a build system other than Gradle, you must configure it to do two tasks.
+If either of these is not done, the checker will not issue any errors on Lombok code.
+\begin{itemize}
+\item set Lombok configuration option \<lombok.addLombokGeneratedAnnotation = true>
+\item delombok the code before passing it to the checker
+\end{itemize}
+
+
+\sectionAndLabel{Specifying your code}{called-methods-spec}
+
+The Called Methods Checker reads method specifications (contracts) that
+state what a method requires when it is called.  It warns if method
+arguments do not satisfy the method's specification.
+
+If you use AutoValue or Lombok, most specifications are automatically
+inferred by the Called Methods Checker, from field annotations such as
+\<@Nullable> and field types such as \<Optional>. See the
+section on defaulting rules for Lombok and AutoValue for more details
+(Section~\ref{called-methods-framework-details}).
+
+In some cases, you may need to specify your code. You do so by writing one of the following type
+annotations:
+\begin{description}
```

Comment:
The manual should list all the annotations.  (It lacks the bottom annotation.)  It should also show the subtyping relationship.  Please draw a type hierarchy diagram like those in other manual sections.